### ISSUE TITLE
Excessive paragraph spacing 

### ISSUE THREAD

**Issue by reimerd** [crosswire/xiphos#927]
I am using Xiphos 4.0.7 (gtk3 webkit1) under Ubuntu 18.04 LTS.

This isn't a "bug", more an enhancement request, or at least a plea for guidance. :) This is the "opposite", in a sense, of #530 in that the paragraphs are well recognized, but the spacing between them is exceedingly generous. Here's an example (ESV module):

![screenshot](https://i.imgur.com/OE0NEiMm.png)

That adds up to a lot of "white space" (or green, in my case) on the screen. The paragraph space area is marked by red boxes -- it could easily be halved (or more), getting more text on the screen (my main goal) while respecting and displaying the paragraph divisions.

I have poked around the `/usr/share/xiphos` directory to see if I could spot where this "CSS" type property is set, but couldn't find it. Ideally, it would be great to have this as a property users could set in their **Preferences** -- one thought being to rename the "Fonts" to "Display", and adding this (and perhaps other) properties for user configuration.

This is hardly a priority item though :D ! Perhaps, then, there could at least be some information about how to modify this locally, even if it means a bit of hacking (short of re-compiling which would be quite beyond me).

Thanks for the app, and for your help with this!

**Comment by LAfricain** [comment]
I use Xiphos 4.1 and  ESV2001 module and the spacing are correct. Update first the module, then try with the last xiphos version.
![spacing](https://user-images.githubusercontent.com/19690477/55177993-d144f100-5184-11e9-9cbe-22bbdad1d741.png)

The CSS, I think is in the `.xiphos` home folder. If not create it. But I'm not sure you can use it with the 4.0.7

**Comment by reimerd** [comment]
Thanks for those suggestions, @LAfricain. I have just tried to follow-up now. And I remember why I'm using 4.0.7 on Ubuntu 18.04: it's what is available in `apt` installation. Sadly, I see this in the [4.1.0 tag release][1]: 

> No longer distributing Linux builds because no one downloads them.

I'm not sure exactly what @karlkleinpaste has in mind: I expect there aren't downloads because the package has been available in a PPA, but this [has disappeared][2] = #932 and people are having difficulty building on Ubuntu (#914 and #928 but that one with more success).

I would certainly download an Ubuntu package if one was available; the Xiphos download page still [points to the PPA which now 404s][3].

It would be great to get 4.1.0 (and subsequent updates), but it's not clear to me how I can best do that! It might mean simply staying with 4.0.7, which is a wee shame, seeing all the improvements incorporated into the latest version.

[1]: https://github.com/crosswire/xiphos/releases/tag/4.1.0
[2]: https://github.com/crosswire/xiphos/issues/932
[3]: http://xiphos.org/download/

**Comment by LAfricain** [comment]
Currently the only way is to compile sword 1.8 and xiphos. For Xiphos with the new Cmake compiler it is easier to compile and at the end it create automaticly a deb package.

**Comment by Bohooslav** [comment]
Can I make in this way rpm package?

**Comment by LAfricain** [comment]
rpm I don't know. But if you compile it will be installed on your system. For deb the other way is to download all the deb package on https://packages.ubuntu.com/disco/xiphos, you need to found all dependencies.

**Comment by LAfricain** [comment]
This the disco packages you need to download on disco :+1:
libbiblesync1.1_1.2.0-1_amd64.deb
libicu63_63.1-6_amd64.deb
libsword-1.8.1_1.8.1+dfsg-8build1_amd64.deb
libsword-common_1.8.1+dfsg-8build1_all.deb
xiphos_4.1.0.1+dfsg1-1_amd64.deb
xiphos-data_4.1.0.1+dfsg1-1_all.deb

**Comment by reimerd** [comment]
@LAfricain Of course, if there's a `.deb`, it's even easier, since you can use the [GDebi Package Installer][1] to get all the dependencies [automatically][2]. But probably you know that already. :) 

Now if only there was a `.deb` for the latest Xiphos, we'd be good to go!


[1]: https://launchpad.net/gdebi
[2]: https://itsfoss.com/gdebi-default-ubuntu-software-center/

**Comment by LAfricain** [comment]
> to get all the dependencies automatically

Not all, it is for that I gave you the list. You need an other version of libicu63 and libbiblesync. After update check the bible and see if the issue is still relevant.

**Comment by LAfricain** [comment]
You can now use my [ppa](https://launchpad.net/~lafricain79/+archive/ubuntu/lafricainenvrac) for Ubuntu.

**Comment by reimerd** [comment]
@LAfricain wrote:

> You can now use my `ppa` for Ubuntu.

Thanks for this! I tried it, but ran into a problem. After updating my software sources and installing Xiphos `amd64 4.1.0.1-1`, launching it silently fails. I get this error message when attempting to launch it from the terminal:

`xiphos: symbol lookup error: xiphos: undefined symbol: _ZN9BibleSync16TransmitInternalEcNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_S5_S5_S5_`

- I also tried re-installing from the .deb from [the one supplied in the package][1], but got the same result.
- I tried to find a place to register this at your PPA's [bugs][b] and [answers][a] pages, but couldn't find any place to add anything! ([This is my][d] identity in launchpad.) **I know this is the wrong place** for it, but I couldn't find the *right* place! Sorry about that.

(I'm on Ubunutu 18.04 LTS.)

[a]: https://answers.launchpad.net/~lafricain79
[b]: https://bugs.launchpad.net/~lafricain79
[d]: https://launchpad.net/~dajare
[1]: https://launchpad.net/~lafricain79/+archive/ubuntu/lafricainenvrac/+sourcepub/10220324/+listing-archive-extra

**Comment by LAfricain** [comment]
Just remove all sword and xiphos related package or compilation. Then clean the $HOME folder. Then try to reinstall. And check during installation biblesync or icu is install too.

**Comment by reimerd** [comment]
@LAfricain - thanks for the reply. Here's what I did:

- `[✓]` removed all sword and xiphos related package or compilation;
- `[✓]` cleaned/emptied the $HOME xiphos/sword folder(s)
- `[✓]` reinstalled from ppa
- `[✓]` checked for biblesync or libicu is installed to ⇒ via `apt-cache search [package]`

And I still get the same error message. :( 

If there is a better place to communicate about this (and thank you again for engaging!) please point me in the right direction.

**Comment by LAfricain** [comment]
My version of libicu is libicu63. Do you have the same? Did clean all the compiled stuff?

**Comment by reimerd** [comment]
@LAfricain 

> My version of libicu is libicu63. Do you have the same?

Looks like I have **libicu60**:

```
$ apt-cache search libicu
icu-devtools - Development utilities for International Components for Unicode
libicu-dev - Development files for International Components for Unicode
libicu60 - International Components for Unicode
libicu-le-hb-dev - ICU Layout Engine API on top of HarfBuzz shaping library (development)
libicu-le-hb0 - ICU Layout Engine API on top of HarfBuzz shaping library
libiculx60 - International Components for Unicode
libicu4j-4.2-java - Library for Unicode support and internationalization
libicu4j-4.4-java - Library for Unicode support and internationalization
libicu4j-49-java - Library for Unicode support and internationalization
libicu4j-java - Library for Unicode support and internationalization
libicu4j-java-doc - Library for Unicode support and internationalization - Docs
```
> Did clean well all the compiled stuff?

As well as I know how to do! Xiphos is the only Sword project related application I have on my system.

> Does the official repo xiphos work?

So, I removed your PPA, `removed` xiphos and xiphos-data, then installed Xiphos from official packages, which gives me this:

```
$ apt search xiphos

bibledit-xiphos/bionic 1.1.1-2build1 amd64
  Glue between bibledit and xiphos

diatheke/bionic 1.7.3+dfsg-9.1build2 amd64
  command line bible browsing and search tool

xiphos/bionic,now 4.0.7+dfsg1-1build2 amd64 [installed]
  environment for Bible reading, study, and research

xiphos-data/bionic,bionic,now 4.0.7+dfsg1-1build2 all [installed,automatic]
  data files for Xiphos Bible study software
```

And, yes—it **does** work. Anything else I can try?

**Comment by LAfricain** [comment]
Ok The only I have idea is to upgrade libicu to libicu63. If it is the problem I will update the ppa.  [ubuntu.package](https://packages.ubuntu.com/disco/xiphos) shows the the libicu63 is needed.
Strange in my system i had no issue?
Then download the disco package https://packages.ubuntu.com/disco/libicu63, reinstall xiphos with the ppa.

**Comment by reimerd** [comment]
**Success**! I installed `libicu63` from the `.deb` file at the link you provided. I now have both libicu63 *and* libicu60 on my system, it seems. (shrug!)

But I then removed my "fresh" official install, added back your PPA, installed ... and it works! Thanks for your patience. It's great to get the latest Xiphos installed! Hopefully this will be a help for others, if the problem surfaces for LTS users.
